src/basics.md

```diff
@@ -1770,6 +1771,48 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-read-environment-variable]: #ex-read-environment-variable
+<a name="ex-read-environment-variable"></a>
+## Read environment variable
+[![std-badge]][std] [![cat-filesystem-badge]][cat-filesystem]
+
+Set an environment variable and read it back. If the variable is not found, then fallback
+to a default value.
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+use std::env;
+use std::path::PathBuf;
+
+# error_chain! {
+#     foreign_links {
+#         Utf8(std::str::Utf8Error);
+#         AddrParse(std::net::AddrParseError);
+#     }
+# }
+
+fn set_env_variable() {
+  env::set_var("CONFIG_FILE", "config.toml");
+}
+
+fn run() -> Result<()> {
+    set_env_variable();
+
+    let default_config_file = "config.toml";
+
+    let config_file_path = match env::var("CONFIG_FILE") {
```

Comment:
This might a matter of taste but how about rewriting this simple transformation from match to `Result` combinators with something like: 
```rust
    let config_file_path = env::var("CONFIG_FILE")
        .map(PathBuf::from)
        .unwrap_or_else(|_| default_config_file.into());
```